api-gateway/src/main/java/com/petclinic/bffapigateway/presentationlayer/v2/InventoryController.java

```diff
@@ -159,6 +159,27 @@ public Mono<ResponseEntity<InventoryResponseDTO>> addInventory(@RequestBody Inve
                 .defaultIfEmpty(ResponseEntity.badRequest().build());
     }
 
+    @SecuredEndpoint(allowedRoles = {Roles.ADMIN, Roles.INVENTORY_MANAGER})
+    @GetMapping(value = "/{inventoryId}/products/search", produces = MediaType.APPLICATION_JSON_VALUE)
+    public Mono<ResponseEntity<Flux<ProductResponseDTO>>> searchProducts(
+            @PathVariable String inventoryId,
+            @RequestParam(required = false) String productName,
+            @RequestParam(required = false) String productDescription) {
+
+        Flux<ProductResponseDTO> products = inventoryServiceClient
+                .searchProducts(inventoryId, productName, productDescription);
+
+        return products
+                .hasElements() // Check if any elements are present in the flux
+                .flatMap(hasElements -> {
+                    if (hasElements) {
+                        return Mono.just(ResponseEntity.ok(products));
+                    } else {
+                        return Mono.just(ResponseEntity.notFound().build());
+                    }
+                });
+    }
+
```

Comment:
You added your endpoint to the InventoryController class which is great because it allows us to find all our endpoints for our respective microservice inside one class. This way it makes the code more organized instead of having every endpoint in the BFFApiGatewayController.